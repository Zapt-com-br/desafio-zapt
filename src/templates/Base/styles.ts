import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 100vh;
`

export const SectionHeader = styled.section`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;
    min-height: 6.4rem;
    z-index: ${theme.layers.menu};

    position: sticky;
    top: 0;
    left: auto;
    right: 0;

    background-color: ${theme.colors.white};
    border-bottom: 1px solid rgba(0, 0, 0, 0.12);
  `}
`

export const Logo = styled.img`
  width: 10rem;
  height: 3rem;
`

export const Content = styled.main`
  ${({ theme }) => css`
    padding-top: calc(${theme.grid.gutter} * 2);
  `}
`

export const SectionFooter = styled.section`
  ${({ theme }) => css`
    padding-top: calc(${theme.grid.gutter} * 2);
  `}
`

export const Footer = styled.p`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;

    padding: ${theme.grid.gutter} 0;
    min-height: 12rem;

    color: ${theme.colors.gray};
  `}
`
