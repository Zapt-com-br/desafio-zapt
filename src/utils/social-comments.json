{
  "title": "O que est√£o falando nas redes sociais",
  "comments": [
    {
      "id": 1,
      "comment": "N√£o da para parar de comer.. esse final de semana pedi 20 pasteis da Ilma e acabaram TODOS em um s√≥ dia... O pessoal aqui de casa ficou viciado de um n√≠vel que terei que comprar toda semana... Assim n√£o tem academia que secura rsrsrsrs Obrigado Zapt.",
      "name": "ferri-gui",
      "image": "/images/social-comments/ferri-gui.svg"
    },
    {
      "id": 2,
      "comment": "Estava precisando achar algum lugar que vendia produtos que v√™em direto da fazenda... a Zapt salvou.. e ainda paquei um pre√ßo muito mais barato que supermecado üôå",
      "name": "paolaC",
      "image": "/images/social-comments/paola.svg"
    },
    {
      "id": 3,
      "comment": "Todos os vinhos da vin√≠cula Bueno est√£o dispon√≠veis no catalogo da Zapt a um pre√ßo incriv√©l... n√£o percam essa oportunidade...",
      "name": "Galv√£o",
      "image": "/images/social-comments/galvao.svg"
    },
    {
      "id": 4,
      "comment": "Nunca tinha achado os produtos na Lush depois que ela saiu do Brasil.. s√≥ a Zapt para conseguir importalos a pre√ßo mais barato do que vendiam aqui! Muito obrigado, recomendo sempre a todos os amigos.",
      "name": "LiciaR",
      "image": "/images/social-comments/liciar.svg"
    },
    {
      "id": 5,
      "comment": "Ontem chegou uma pizza que comprei na Zapt e me lembrou as que minha av√≥ fazia... sensacional!",
      "name": "shitz",
      "image": "/images/social-comments/shitz.svg"
    },
    {
      "id": 6,
      "comment": "Nossa cada dia tem uma oferta mais legal do que a outra.. Al√©m de produtos de qualidade o pr√©co √© imbat√≠vel",
      "name": "rafael",
      "image": "/images/social-comments/rafael.svg"
    },
    {
      "id": 7,
      "comment": "Achei uma promo√ß√£o incr√≠vel de roupinha de beb√™, foi o que salvou o enxoval do Pedro.",
      "name": "gust",
      "image": "/images/social-comments/gust.svg"
    },
    {
      "id": 8,
      "comment": "Meu dono finalmente conseguiu achar uma comida que eu gosto! Agora vou conseguir engordar e ser feliz.",
      "name": "Cookie",
      "image": "/images/social-comments/cookie.svg"
    },
    {
      "id": 9,
      "comment": "N√£o conhecia a Zapt.. um amigo meu indicou... falou para eu dar uma olhada nos pre√ßos das cervejas artesanais.. incr√≠vel.",
      "name": "duque",
      "image": "/images/social-comments/duque.svg"
    }
  ]
}
