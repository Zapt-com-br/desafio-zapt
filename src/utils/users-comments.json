{
  "title": "O que as pessoas estão achando",
  "comments": [
    {
      "id": 1,
      "image": "/images/user-comments/image-1.png",
      "comment": "Não consigo mais viver sem os produtos do Sítio, a diferença do que eu comprava antes e agora é abissal",
      "user": {
        "avatar": "/images/user-comments/patricia.svg",
        "name": "Patricia Godoy",
        "city": "São Paulo",
        "state": "São Paulo"
      }
    },
    {
      "id": 2,
      "image": "/images/user-comments/image-2.png",
      "comment": "Depois que você começa a usar produtos de qualidade é bem difícil voltar atrás...",
      "user": {
        "avatar": "/images/user-comments/marcinho.svg",
        "name": "Marcinho Ribeiro",
        "city": "Belo Horizonte",
        "state": "Minas Gerais"
      }
    },
    {
      "id": 3,
      "image": "/images/user-comments/image-1.png",
      "comment": "Não consigo mais viver sem os produtos do Sítio, a diferença do que eu comprava antes e agora é abissal",
      "user": {
        "avatar": "/images/user-comments/patricia.svg",
        "name": "Patricia Godoy",
        "city": "São Paulo",
        "state": "São Paulo"
      }
    },
    {
      "id": 4,
      "image": "/images/user-comments/image-2.png",
      "comment": "Depois que você começa a usar produtos de qualidade é bem difícil voltar atrás...",
      "user": {
        "avatar": "/images/user-comments/marcinho.svg",
        "name": "Marcinho Ribeiro",
        "city": "Belo Horizonte",
        "state": "Minas Gerais"
      }
    }
  ]
}
