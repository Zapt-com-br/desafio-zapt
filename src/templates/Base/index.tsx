import * as S from './styles'

export type BaseTemplateProps = {
  children: React.ReactNode
}

const Base = ({ children }: BaseTemplateProps) => (
  <>
    <S.SectionHeader>
      <S.Logo src="/img/logo.svg" alt="Imagem do logo da Zapt" />
    </S.SectionHeader>
    <S.Wrapper>
      <S.Content>{children}</S.Content>
      <S.SectionFooter>
        <S.Footer>Desafio Frontend Zapt 🧑‍💻</S.Footer>
      </S.SectionFooter>
    </S.Wrapper>
  </>
)

export default Base
