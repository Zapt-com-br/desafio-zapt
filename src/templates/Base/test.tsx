import { render, screen } from 'utils/test-utils'

import Base from '.'

describe('<Base />', () => {
  it('should render base template with logo and footer', () => {
    const { container } = render(
      <Base>
        <h1>Base Template</h1>
      </Base>
    )

    expect(
      screen.getByRole('img', { name: /imagem do logo da zapt/i })
    ).toHaveAttribute('src', '/img/logo.svg')

    expect(
      screen.getByRole('heading', { name: /base template/i })
    ).toBeInTheDocument()

    expect(screen.getByText(/desafio frontend zapt üßë‚Äçüíª/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
