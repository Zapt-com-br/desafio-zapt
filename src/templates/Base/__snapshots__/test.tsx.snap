// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Base /> should render base template with logo and footer 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 6.4rem;
  z-index: 20;
  position: fixed;
  top: 0;
  left: auto;
  right: 0;
  width: 100%;
  background-color: #ffffff;
  border-bottom: 1px solid rgba(0,0,0,0.12);
}

.c1 {
  width: 10rem;
  height: 3rem;
}

<section
  class="c0"
>
  <img
    alt="Imagem do logo da Zapt"
    class="c1"
    src="/img/logo.svg"
  />
</section>
`
